text/0000-wallet-replay-protection.md

```diff
@@ -0,0 +1,298 @@
+
+The goal of this RFC is to propose new wallet rules that, when strictly followed, protect the user from all known malicious (re)play attacks. This is done by changing the transaction building process but at the same time keeping it configurable enough to still allow the user to take control of their privacy. The solution mostly supports all the transaction building flows that were used before while encouraging default use of PayJoin transaction with some probability. Contributing inputs on to a 'receive' transaction can be turned off in the wallet configuration settings while still keeping the protection against replay attacks. The wallet also allows choosing different wallet configurations for different use cases e.g. a withdrawal from an exchange never does a PayJoin transaction. 
+
+# Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+As we have already mentioned, a transaction can be replayed if the inputs are recreated and its outputs don't exist. One way to stop this attack would be to make it impossible to recreate an input. Of course the owner can always recreate it, but the problem is really if anyone else recreates it by replaying the transaction that created the input. Luckily, there is a way to make an input that is protected and can't be recreated by anyone else except by the owner. This seems impossible because you can always just go one transaction back and recreate the outputs that created the inputs and keep repeating this until you hit a transaction that can be replay at which point you can replay all the next transactions and thus successfully recreate the target input. To prevent this, we create a 'stopping' transaction `T` for which we _know_ it can't be recreated. We achieve this by creating an output in `T` that we call an `anchor` output which will never be spent. Since Grin does not allow duplicate outputs, it means that the transaction that created `anchor` output can't be recreated because it would attempt to insert a duplicate output in the UTXO set which is invalid by Grin rules. Along with an `anchor` output we create additional outputs `O1` and `O2` in `T` which we call `Protected` outputs - it's not obvious why they have this name yet.
+
+Let's say we have a transaction `T`
+```
+I1  -> anchor
+    -> O1
+    -> O2
+```
+
+We now use `O2` in a new transaction `T2` as an input to send money to Bob.
+
+```
+O2  -> O3 (change output)
+    -> Bob_output
+```
+
+The key idea here is to observe that `T2` can't be replayed by anyone else. Only we can create the input needed for the transaction to be replayed `O2`. The attacker could attempt to replay the previous transaction `T` to create the  `O2` input, but this would only be possible if they managed to replay transaction `T` which can't be replayed because it has an `anchor` output that has not been spent and hence can't be valid. This means that our transaction `T2` is protected against other parties replaying it. 

```

Comment:
The part "but this would only be possible if they managed to replay transaction `T` which" is redundant and can be replaced by "but T"